### PR TITLE
update k8s.io dependencies to v0.28.1

### PR DESCRIPTION
**Pull request by fjogeleit**
To update the dependencies of https://github.com/kyverno/playground we need to to use k8s.io/apiextensions-apiserver v0.28.1

In case this project also wants to do this update soon, I would like to provide my changes as a contribution.

### CODE REVIEW

**Comment by eddycharly** [comment]
Hmm, not using prow...

File: pkg/validatorfactory/factory.go [hunk 1]
```diff
@@ -43,12 +52,12 @@ func (v *ValidatorEntry) IsNamespaceScoped() bool {
...
 }
 
-func (v *ValidatorEntry) SchemaValidator() *validate.SchemaValidator {
+func (v *ValidatorEntry) SchemaValidator() validation.SchemaValidator {
 	if v.schemaValidator != nil {
 		return v.schemaValidator
 	}
 
-	v.schemaValidator = validate.NewSchemaValidator(v.Schema, nil, "", strfmt.Default)
+	v.schemaValidator = &BasicValidatorAdapter{SchemaValidator: validate.NewSchemaValidator(v.Schema, nil, "", strfmt.Default)}
```
Comment (eddycharly) [inline pkg/validatorfactory/factory.go:60]: @apelisse @alexzielenski do we need to create a validator adapter ?

cc @fjogeleit

**Comment by eddycharly** [comment]
This would be awesome to have this in !

File: pkg/validatorfactory/factory.go (hunk 1 above)
Comment (fjogeleit) [reply pkg/validatorfactory/factory.go:60]: The interface of the Validator changed in https://pkg.go.dev/k8s.io/apiextensions-apiserver@v0.28.2/pkg/apiserver/validation#SchemaValidator

They do something similar:

https://github.com/kubernetes/apiextensions-apiserver/blob/v0.28.2/pkg/apiserver/validation/validation.go#L46

File: pkg/validatorfactory/factory.go [hunk 2]
```diff
@@ -26,12 +27,20 @@ type ValidatorFactory struct {
 	validatorCache map[schema.GroupVersionKind]*ValidatorEntry
 }
 
+type BasicValidatorAdapter struct {
```
Comment (eddycharly) [inline pkg/validatorfactory/factory.go:30]: shall we make this type private ?

Comment (fjogeleit) [reply pkg/validatorfactory/factory.go]: done

**Comment by alexzielenski** [comment]
thanks for this!

/lgtm
/approve

**Pull request merged**